pkg/cmd/validate.go

```diff
@@ -263,13 +263,9 @@ func ValidateFile(filePath string, resolver *validatorfactory.ValidatorFactory)
 		return field.Invalid(field.NewPath("metadata"), nil, err.Error())
 	}
 
-	//!TODO: source this information from OpenAPI somehow
-	crdIsNamespaceScoped := true
-
-	// Infer namespace scoped based on presence of namespace field in user data
-	// for now :(
-	if n := obj.GetNamespace(); len(n) == 0 {
-		crdIsNamespaceScoped = false
+	isNamespaced := validators.IsNamespaceScoped()
+	if isNamespaced && obj.GetNamespace() == "" {
+		obj.SetNamespace("default")
```

Comment:
Is this necessary? I'm curious if something else in the k8s tooling handles this logic

Reply:
Here's a note: this is typically going to be populated with default, OR the namespace defined in the kubectl context, or the namespace given in the kubectl command, so it's somewhat hard to predict whether it'll be "default". Should we have a param to kubectl-validate that defaults to "default", but also can use the value from the context if that's used?

Reply:
Shall I add a `TODO` and create an issue to track that ? (and keep this PR only about deducing the scope of the schema)